sway-core/src/semantic_analysis/module.rs

```diff
@@ -299,10 +299,19 @@ impl ty::TyModule {
                         .chain(all_nodes.iter().flat_map(|x| x.contract_fns(engines)))
                         .collect::<Vec<_>>();
 
+                    // collect the fallback fn
+                    let fallback_fn = all_nodes.iter().find_map(|x| match &x.content {
+                        ty::TyAstNodeContent::Declaration(ty::TyDecl::FunctionDecl(decl)) => {
+                            let d = engines.de().get(&decl.decl_id);
+                            d.is_fallback().then_some(decl.decl_id)
+                        }
+                        _ => None,
+                    });
```

Comment:
We need to handle the case where the user mistakenly marks multiple functions with `#[fallback]` and throw an error for it.